File: Editorial/137_Single_Number_II/C/5_Equation_for_Bitmask.c
int singleNumber(int* nums, int numsSize) {
    // Initialize seenOnce and seenTwice to 0
    int seenOnce = 0, seenTwice = 0;

    // Iterate through nums
    for (int i = 0; i < numsSize; i++) {
        // Update using derived equations
        seenOnce = (seenOnce ^ nums[i]) & (~seenTwice);
        seenTwice = (seenTwice ^ nums[i]) & (~seenOnce);
    }

    // Return integer which appears exactly once
    return seenOnce;
}

File: Editorial/137_Single_Number_II/Go/5_Equation_for_Bitmask.go
func singleNumber(nums []int) int {
    // Initialize seenOnce and seenTwice to 0
    seenOnce, seenTwice := 0, 0

    // Iterate through nums
    for _, num := range nums {
        // Update using derived equations
        seenOnce = (seenOnce ^ num) & (^seenTwice)
        seenTwice = (seenTwice ^ num) & (^seenOnce)
    }

    // Return integer which appears exactly once
    return seenOnce
}
